src/history.h

```diff
@@ -40,9 +40,13 @@ struct io_streams_t;
 typedef std::vector<wcstring> path_list_t;
 
 enum history_search_type_t {
-    // The history searches for strings containing the given string.
+    // We don't know what type of search to perform.
+    HISTORY_SEARCH_TYPE_UNKNOWN,
```

Comment:
> Incidentally I have no objection if you want to start enums at 1 instead of 0.

Doing so obviously doesn't ensure uninitialized enum vars aren't misinterpreted but it does reduce the likelihood. Even better would be picking a random, distinct, value for the value of the first enum symbol when the exact value doesn't otherwise matter. My experience debugging problems the past three decades has shown that using zero or one for the first value is problematic. For example, my $dayjob project was using one as the initial value for a bunch of IDs stored in the transitory database used by the unit tests. Which masked the fact that some of the tests were incorrectly using a "abc" ID where they should have been using a "xyz" ID. Because both ID name spaces were initialized to one and incremented in lock step the mistake was masked. Randomizing the initial value for the ID name spaces exposed those mistakes in the unit tests as well as a bug in the code being tested.